ssz-rs/src/merkleization/merkleize.rs

```diff
@@ -286,54 +282,100 @@ impl std::fmt::Debug for Tree {
 // Return the full Merkle tree of the `chunks`.
 // Invariant: `chunks.len() % BYTES_PER_CHUNK == 0`
 // Invariant: `leaf_count.next_power_of_two() == leaf_count`
-// NOTE: naive implementation, can make much more efficient
-pub fn compute_merkle_tree(
-    hasher: &mut Sha256,
-    chunks: &[u8],
-    leaf_count: usize,
-) -> Result<Tree, Error> {
+pub fn compute_merkle_tree(chunks: &[u8], leaf_count: usize) -> Result<Tree, Error> {
```

Comment:
Since the hashes is only used in `mix_in_decoration()` to calculate a pair of hashes I removed the hasher here to simplify the API